adyenv6core/src/com/adyen/commerce/services/impl/DefaultAdyenRequestService.java

```diff
@@ -0,0 +1,132 @@
+
+import com.adyen.commerce.services.AdyenRequestService;
+import com.adyen.model.checkout.CardDetails;
+import com.adyen.model.checkout.PaymentRequest;
+import de.hybris.platform.commercefacades.order.data.CartData;
+import de.hybris.platform.commercefacades.order.data.OrderEntryData;
+import de.hybris.platform.core.model.order.CartModel;
+import de.hybris.platform.order.CartService;
+import de.hybris.platform.servicelayer.config.ConfigurationService;
+import de.hybris.platform.store.services.BaseStoreService;
+import org.apache.commons.lang3.StringUtils;
+import org.spockframework.util.CollectionUtil;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+public class DefaultAdyenRequestService implements AdyenRequestService {
+
+
+    private BaseStoreService baseStoreService;
+    private CartService cartService;
+    private ConfigurationService configurationService;
+
+    private static final String L2L3_EDS_SUPPORTED_BRANDS = "adyen.l2l3eds.supported.brands";
+    private static final String L2L3_EDS_SUPPORTED_COUNTRIES = "adyen.l2l3eds.supported.countries";
+
+    public DefaultAdyenRequestService(BaseStoreService baseStoreService, CartService cartService, ConfigurationService configurationService) {
+        this.baseStoreService = baseStoreService;
+        this.cartService = cartService;
+        this.configurationService = configurationService;
+    }
+
+    @Override
+    public void populateL2L3AdditionalData(final Map<String, String> additionalData, final CartData cartData) {
+        // required fields
+        if (cartData.getTotalTax() != null) {
+            additionalData.put(TOTAL_TAX_AMOUNT, String.valueOf(cartData.getTotalTax().getValue()));
+        }
+        if (cartData.getMerchantCustomerId() != null) {
+            additionalData.put(CUSTOMER_REFERENCE, cartData.getMerchantCustomerId());
+        }
+        // not required but available
+        if (cartData.getDeliveryCost() != null) {
+            additionalData.put(FREIGHT_AMOUNT, String.valueOf(cartData.getDeliveryCost().getValue()));
+        }
+        if (cartData.getDeliveryAddress() != null) {
+            if (cartData.getDeliveryAddress().getPostalCode() != null) {
+                additionalData.put(DESTINATION_POSTAL_CODE, cartData.getDeliveryAddress().getPostalCode());
+            }
+            if (cartData.getDeliveryAddress().getCountry() != null && cartData.getDeliveryAddress().getCountry().getIsocode() != null) {
+                additionalData.put(DESTINATION_COUNTRY_CODE, cartData.getDeliveryAddress().getCountry().getIsocode());
+            }
+        }
+
+        // Extract item details from cartData and populate additionalData using a stream
+        if (cartData.getEntries() != null) {
+            IntStream.range(0, cartData.getEntries().size()).forEach(i -> {
```

Comment:
Wouldn't it be better to have IntStream.range(1, cartData.getEntries().size()+1)?